### ISSUE TITLE
Watch mode / `build --watch`

### ISSUE THREAD

**Issue by felipecrs** [docker/buildx#1330]
Since Buildkit knows exactly which files in the system are required for the build, a `--watch` mode, that would rebuild an image automatically in case one of the files depended by it changes would come very in handy when developing docker images locally, or especially when testing them.

When called with `--watch`, buildx would NOT exit upon creation of the image, but instead would wait until the user exits with <kbd>Ctrl</kbd>+<kbd>C</kbd>.

Additionally, a `bake --watch` would be even better.

**Comment by sprat** [comment]
I agree, a `bake --watch` would be awesome!

**Comment by jsternberg** [comment]
I don't believe this is something that we will work on but if someone is interested in doing this feel free to ask any questions on this issue about how it might be implemented. I'll leave this open for anyone interested in trying to contribute this but I also don't think this issue is easy to complete and I wanted to set realistic expectations.
